src/rezplugins/shell/cmd.py

```diff
@@ -182,12 +184,34 @@ def _create_ex():
             executor.interpreter._saferefenv('REZ_ENV_PROMPT')
             executor.env.REZ_ENV_PROMPT = literal(newprompt)
 
+        # Make .py launch within cmd without extension.
+        if self.settings.additional_pathext:
+            executor.command('set PATHEXT=%PATHEXT%;{}'.format(
```

Comment:
It's a fairly minor point, but in nested rez-envs this will cause PATHEXT to become `>PY;.PY;.PY...`. If it's easy, it may be worth updating this command to only add the additional pathext if it isn't already in PATHEXT. If it's a pain I wouldn't worry too much.

Reply:
Actually, note also that wrt #707 (see my comments there), that with sufficient env-var control added to the project, this special case may not be necessary at all. I'm imagining something like:

```
env_var_settings = {
  "PATHEXT": {
    "set_always": ".PY",  # always set PATHEXT=.PY in a new shell...
    "path_modify_mode": "retain_after"  # ..but also keep pre-rez value of PATHEXT appended as well
  }
}
```

Reply:
I personally can wait to have #707 done first. Everyone who has a need for better shell support may feel free to merge this branch for testing.
Up to you - I will fix the issue above as suggested, just in case.

Reply:
For now I've implemented de-duplication as is for cmd and PowerShell.
I will move the discussion about the generic variable handling to the #707